### PR TITLE
Blazor byte array support

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #22444
Addresses #22045

### CODE REVIEW

**Comment by guardrex** [comment]
Ahoy @TanayParikh! 🚢 ... I'm **_drafting_** the doc sections with examples for byte array interop:

* This is only a **_draft_** for sure because I'm not on preview bits yet. I generally try to avoid the 🙉 *Core Churn*&trade; 🙉 until we get to RC1, then I go on the new bits and suspend my local work in released VS in favor of .NET CLI until the new VS releases at GA. Anyway, it just results in my needing to ask on a few points that I can't test and work out locally.
* We have the two topics for interop: *Call .NET from JS* and *Call JS from .NET*. For these new sections+examples, I'd like to have:
  * The byte array go .NET to JS with a decoded string come back for the *Call JS* topic.
  * The byte array go JS to .NET with a decoded string (the same string actually) come back for the *Call .NET* topic.
  
  In each case, the string is displayed to the dev, either in the component in the *Call JS* topic or via an `alert()` in the JS in the *Call .NET* topic.
  
  This layout of the coverage makes things clean and organized for the coverage with minimal reader confusion on the direction that the byte array is traveling (*... and minimal doc author confusion, too!* 😆).

Can you put an 👁️ on this PR at this early stage and see where the code is going to need a touch-up to make it work? If not, no worries. We can hold this PR in a draft state until I go on RC1 and then I'll work the 🐞 out of the code later.

File: aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:417]:
```suggestion
  function sendByteArray() {
```

File: aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:422]: Perhaps

```suggestion
    DotNet.invokeMethodAsync('BlazorSample', 'ReceiveByteArray', data)
```

to keep consistent with other examples? Otherwise they'll have to create a new assembly this for this.

File: aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:439]:
```suggestion
    <button onclick="sendByteArray()">Send Bytes</button>
```

File: aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md:656]:
```suggestion
  function receiveByteArray(bytes) {
```

File: aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:447]: Can this be removed?

```suggestion
```

File: aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md:685]: Can this be removed?

```suggestion
```

**Review by TanayParikh — COMMENTED** [review]
Thanks @guardrex for putting this together. This looks great! 

Left some general comments, still need to put it in a project and double check it compiles / works as expected. (I'll try to get to this, this week).

**Comment by guardrex** [comment]
Thanks @TanayParikh ...

* Yes, I do use `BlazorSample` ... that was in here because that sample app is sitting out on my desktop waiting for me to adopt 6.0 locally. I fixed it. Eventually, this code will go into the snippet sample apps. Those are named `BlazorSample`.
* Fixed the function/method naming to `sendByteArray`/`SendByteArray`.
* We use arrow functions across the Blazor docs because the PU engineers usually send over their JS functions that way. I imagine that they like it because it's compact.
* The `<p>` content at the bottom has to stay: The quote is from the movie, it's copyrighted, and CELA/Sony would have a fit if we didn't keep the attribution. This is common across the docs. I have all kinds of Star Wars, Star Trek, Dr. Who, etc. things in the docs for fun. 🚀👽

File: aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md
Comment (TanayParikh) [inline aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:453]: It seems using `Encoding.Unicode.GetString` could have platform specific peculiarities. I believe the following should be a bit more general. 

```suggestion
            System.Text.Encoding.UTF8.GetString(receivedBytes, 0, receivedBytes.Length));
```

**Review by TanayParikh — APPROVED** [review]
Thanks @guardrex! Verified the sample code is working as expected (with the following minor change to encoding). Could we go ahead with merging/publishing these docs? The feature was just [announced](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-6/#efficient-byte-array-transfer-in-javascript-interop) and I'm sure users will find this super helpful! 😄 

Also, should we update the menu bar as well? Otherwise, if users just run the app they'll have to manually navigate to the newly created pages. Not sure if there's a general convention for this situation in the docs.

```razor
        <div class="nav-item px-3">
            <NavLink class="nav-link" href="send-byte-array-dotnet-to-js">
                <span class="oi oi-list-rich" aria-hidden="true"></span> .NET to JS
            </NavLink>
        </div>
        <div class="nav-item px-3">
            <NavLink class="nav-link" href="send-byte-array-js-to-dotnet">
                <span class="oi oi-list-rich" aria-hidden="true"></span> JS to .NET
            </NavLink>
        </div>
```

**Comment by guardrex** [comment]
Thanks @TanayParikh ... We aren't showing navigation for every example later in the docs, such as at this point. It takes up too much space for little benefit. Early examples do show it (e.g., tutorials), and the *Routing* doc covers it in the *Fundamentals* node. We let those docs make devs aware of it.

I was going to update the component naming to match the snippet sample app later, but I went ahead and did that now. These examples will be added to the snippet sample apps sometime after GA. The docs don't build preview releases, at least not yet. We might build doc examples for RC.

No need to ping the others. They're all too busy to look at this 🏃😅. We worked out that the responsible engineer gets a single ping on these when the PR goes up. If they don't respond in a week or so, then Artak will look. He'll either be able to approve or reach out to the engineer for a look. In the end, I'm probably going to add a live doc coverage link to the topic section(s) for all of the 6.0 updates for a final management/engineer look before GA. For now tho, I only need your sign-off to merge, so I'll go ahead and get this in.

**Pull request merged**
